include/robot_localization/ros_filter_bias_estimator.h

```diff
@@ -0,0 +1,186 @@
+         */
+        void reset();
+
+        void setDebugInfo(bool debug, bool verbose)
+        {
+            debug_ = debug;
+            verbose_ = verbose;
+        }
+        bool getDebug() const { return debug_; }
+        bool getVerbose() const { return verbose_; }
+
+        /**
+         * @brief Sets the data from a prior estimator
+         * 
+         * @param orientation_estimate Roll/Pitch/Yaw in rad
+         * @param orientation_variance Roll/Pitch/Yaw variance in rad
+         * @param speed Speed in appropriate units
+         * @param time_s Time in seconds
+         */
+        void setImuDynamicCorrectionData(const Eigen::Vector3d &orientation_estimate,
+                                         const Eigen::Vector3d &orientation_variance,
+                                         double speed,
+                                         double time_s);
+        
+        /**
+         * @brief Updates the bias estimate, returns the corrected measurement and associated variance, and determines
+         *        whether the resulting update is valid.
+         * 
+         * @param orientation_measurement Original orientation measurement in, updated/biased measurement out
+         * @param measurement_variance Original variance in, updated/biased variance out
+         * @param sensor_timeout The timeout of the sensor in seconds
+         * @param time_s The current time in seconds
+         * @param orientation_estimate The current estimate of the orientation - used for deviation checking
+         * @param estimate_variance The current variance of the orientation estimate - used for deviation checking
+         * @param is_valid std::vector<bool> whether each axis of the biased orientation measurement and variance are valid for inclusion in the filter
+         * @param debug_stream Used for debug output, if configured
+         */
+        void updateBiasEstimate(Eigen::Vector3d &orientation_measurement,
+                                Eigen::Vector3d &measurement_variance,
+                                double sensor_timeout,
+                                double time_s,
+                                Eigen::Vector3d &orientation_estimate,
+                                Eigen::Vector3d &estimate_variance,
+                                std::vector<bool> &is_valid,
+                                std::ofstream &debug_stream);
+
+        // Parameters setters and getters
+        void set_estimation_axes(std::vector<bool> is_estimating) {
+            for(uint8_t axis = 0; axis < std::min(static_cast<int>(is_estimating.size()), ESTIMATION_AXES); axis++)
+            {
+                estimation_axes_[axis] = is_estimating[axis];
+            }
+        }
+        void get_estimation_axes(std::vector<bool> &is_estimating) const {
+            for(uint8_t axis = 0; axis < ESTIMATION_AXES; axis++)
+            {
+                is_estimating.push_back(estimation_axes_[axis]);
+            }
+        }
```

Comment:
should this have a `return`?

Reply:
Not, it is a fill/return in-place, so just modifies the vector that is provided.

Reply:
Ah, yes, going through too quickly